pkgs/applications/networking/ipfs-cluster/default.nix

```
@@ -2,17 +2,24 @@
 
 buildGoModule rec {
   pname = "ipfs-cluster";
-  version = "0.14.4";
+  version = "0.14.5";
 
-  vendorSha256 = "sha256-4j6aPs17YNXyPIRr5NshAPYIfNM08GlYV13jnGtJzQc=";
+  vendorSha256 = "sha256-ykUjq7Svp3+kUNnFkwsBlC+C4nws6Yvu3bk2Wb4c8vY=";
 
   src = fetchFromGitHub {
     owner = "ipfs";
     repo = "ipfs-cluster";
     rev = "v${version}";
-    sha256 = "sha256-82t3sHMKZiV6sYnW72N94qfRZ/aMkavj+hiAyg5viHQ=";
+    sha256 = "sha256-Xb7QbBmCJKgokxvdbtWxtnNIS/iUsYFLlRzgfoABAq8=";
   };
 
+  # Disable flaky test until https://github.com/ipfs/ipfs-cluster/issues/1586 is fixed
+  postPatch = ''
+    substituteInPlace 'monitor/metrics/window_test.go' --replace \
+      'func TestWindow_Distribution(t *testing.T) {' \
+      'func TestWindow_Distribution(t *testing.T) { t.Skip("Skipping flaky test")'
+  '';
```

> Is there a better way to skip Go tests than patching the source?

Not really, some packages just remove the problematic test.
```suggestion
  postPatch = ''
    rm 'monitor/metrics/window_test.go'
  '';
```

---

I'm inclined to not delete the entire file as there are quite a few other tests in there that don't fail.